packages/office-addin-mock/src/officeMockObject.ts

```diff
@@ -5,9 +5,10 @@ import { isValidError, PossibleErrors } from "./possibleErrors";
  * @param object Object structure to provide initial values for the mock object (Optional)
  */
 export class OfficeMockObject {
-  constructor(object?: ObjectData) {
+  constructor(object?: ObjectData, isOutlook: boolean = false) {
```

Comment:
I'm not sure I like having to put "true" here for all Outlook usage of the mock library.  It would be nice if there were some other way to figure that out and do the right thing automatically.

Reply:
Agree, I will investigate where we can get this information

Reply:
I searched on what options we have and doesn't seem likely that detecting automatically is feasible.
There is a function in `office-addin-manifest` that detects the adding type given a manifest path. 
The problem is that we don't know the manifest path in the tests. If we go into that route, I don't think passing the manifest path is easier than passing an Outlook flag, so I don't think we can detect the adding type automatically.

Reply:
Is there some we could deduce it from the test json? Like is "mailbox" always the top level?

Reply:
We could go and detect this example you said (or similar) and it could cover some cases automatically.
But the current design gives the freedom for the client to mock lower-level objects, instead of having to start from the top layer, so the mailbox might not be present sometimes.

The more I think of it, the less I think it can be done automatically. Maybe we can improve this proposed design if we think of a better way.

Reply:
How would you mock child objects without mocking the parents?

This isn't something we should put out there if we want a better way later since it changes the method signature.  Customer s will put in the boolean for their outlook tests as the documentation tells them, but when we change it, the boolean goes away.  If we're going to require something (boolean in the signature or mailbox or something in the json) then I would prefer the json.

Reply:
"How would you mock child objects without mocking the parents?"
You could just mock some lower-level object right away (like `range`). Outlook doesn't have `sync` so there is no need to start mocking from a high-level object.

I agree that some things in JSON look better.
I think of enforcing the following key:
```
mockJson = {
outlook: true
}```
So this outlook key would have to exist for outlook objects.

Reply:
I'm still not clear on the lower level stuff. 'range' isn't an Outlook object and does require sync.  You couldn't mock userProfile.DisplayName without putting context.mailbox in front of it could you?  My thought is that if outlook requires mailbox at the top (or some limited set of top level objects unique to outlook) then we could figure out it's outlook based on the structure of the json.

If you wanted to add a property to the json (I might be convinced to go with this) to indicate Outlook I wouldn't say "outlook: true" as that is too specific and unique.  I would say something like "host: outlook" which would allow us to use the same filed in the future for other hosts as well (maybe host specific api structure validation would be nice).

Reply:
You could userProfile.DisplayName without putting context.mailbox in front

I agree that outlook: true is too specific. But I am liking the host: outlook idea

Reply:
What's the scenario in which you would have userProfile at the top level of the office.js api in the code you are testing?  We aren't technically preventing in the mock . . . but the mock is for replacing the office.js api calls in tests and so it should match the api structure shouldn't it?

Reply:
it should match the API structure, you just don't need to start from the upper most layer.
You can just mock some lower-level API without having to go through the whole path until you get to it.